src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace_XmlConsumption.cs

```diff
@@ -910,7 +910,13 @@ private static IList<MetadataReference> CreateReferenceList(TestWorkspace worksp
             var references = CreateCommonReferences(workspace, element);
             foreach (var reference in element.Elements(MetadataReferenceElementName))
             {
-                references.Add(MetadataReference.CreateFromFile(reference.Value));
+                // Read the image to an ImmutableArray<byte>, since the GC does a better job of tracking these and
+                // knowing when it's necessary to run finalizers to clean up old Metadata objects that are no longer in
+                // use. There are no public APIs available to directly dispose of these objects, so we are relying on GC
+                // running finalizers to avoid OutOfMemoryException during tests.
```

Comment:
could you explain this a bit more?  (the bit about immutablearrays and whatnot).  i'm not sure i get it.  but i'm fascinated nonetheless!

Reply:
so.... like... isn't this *much* wworse during testing?  doesn't this mean that for each of these, we're always allocating a new byte-array (the size of the file!).  if we have multiple tests that reference htese dlls... aren't we reading in teh file every time and having to rehydrate the metadata, instead of being able to share?

Should we potentially just have a single static map of file-name to metadata-reference for tests?  there can't be that many we reference right?

Reply:
> so.... like... isn't this much wworse during testing? 

No, this is the same as what we were doing before. I just changed the location where allocations were occurring. The difference is now the GC _knows_ we're allocating and thus knows to run the compacting GC to reclaim the memory when it's running low.

Reply:
I updated the comment to clarify that the improvement is relative to `Marshal.AllocHGlobal`.